draft-ietf-quic-version-negotiation.md

```diff
@@ -342,16 +349,20 @@ the Version Information was missing, the client MUST close the connection; if
 the connection was using QUIC version 1, that connection closure MUST use a
 transport error of type `VERSION_NEGOTIATION_ERROR`.
 
-If a client has reacted to a Version Negotiation packet, it MUST validate that
-the server's `Other Versions` field does not contain the client's original
-version, and that the client would have selected the same negotiated version if
-it had received a Version Negotiation packet whose Supported Versions field had
-the same contents as the server's `Other Versions` field. If any of these
-checks fail, the client MUST close the connection; if the connection was using
-QUIC version 1, that connection closure MUST use a transport error of type
-`VERSION_NEGOTIATION_ERROR`. This connection closure prevents an attacker from
-being able to use forged Version Negotiation packets to force a version
-downgrade.
+The client MUST validate the server `Other Versions` field by confirming that it
```

Comment:
This requirement only applies to clients that have reacted to a VN packet right? What validation MUST a client do if this connection didn't involve a VN packet? (in other words, why drop the "If a client has reacted to a Version Negotiation packet," part?)

Reply:
Why shouldn't the client not do the same validation during compatible version negotiation?

Reply:
There are ways in which the two are different, so I think David is right here.

Let's say that there are two groups of versions, the versions in each group compatible with each other, but incompatible with the versions in the other group.  The client prefers a version in the first group, but the client also knows that the versions in the first group are not widely implemented.  As the client doesn't like the added latency of version negotiation, it chooses the second group, which contains a version that is totally OK.  However, if it gets a Version Negotiation packet as a result, it will happily switch to the first group.

So yeah, let's go with the "if VN" condition.